component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -175,6 +170,42 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
         }
     }
 
+    /**
+     * @param userName              certificate's username
+     * @param authenticationContext the authentication context
+     * @param data                  certificate's data
+     * @param claims                claims of the user
+     * @param cert                  X509 certificate
+     * @throws AuthenticationFailedException
+     */
+    private void addOrValidateCertificate(String userName, AuthenticationContext authenticationContext, byte[] data,
+                                          Map<ClaimMapping, String> claims, X509Certificate cert) throws
+            AuthenticationFailedException {
+        boolean isSelfRegistrationEnable = Boolean.parseBoolean(getAuthenticatorConfig().getParameterMap().
+                get(X509CertificateConstants.ENFORCE_SELF_REGISTRATION));
+        if (isSelfRegistrationEnable) {
```

Comment:
Why do we have isSelfRegistrationEnable?

Reply:
Default behavior is allowing any user by registering x509 certificate in user claim (if it is very first login with basic) and validate that self registered certificate when the same user trying to authenticate with X509 certificate.
I have make it as configurable, so that user wanted to do this self registration can use this, otherwise they can simply check whether authenticated user belongs to that certificate.